src/mixins/canvas_events.mixin.js

```diff
@@ -818,13 +827,20 @@
      * @private
      */
     _fireOverOutEvents: function(target, e) {
-      this.fireSyntheticInOutEvents(target, e, {
-        targetName: '_hoveredTarget',
-        canvasEvtOut: 'mouse:out',
-        evtOut: 'mouseout',
-        canvasEvtIn: 'mouse:over',
-        evtIn: 'mouseover',
+      var _this = this, _hoveredTarget = this._hoveredTarget,
+          _hoveredTargets = this._hoveredTargets, targets = this.targets,
+          diff = _hoveredTargets.length - targets.length;
+      [target].concat(targets, new Array(diff > 0 ? diff : 0).fill(null)).forEach(function(_target, index) {
```

Comment:
@jakedowns sorry i did not notice this.
We still support internet explorer and fill is not an option there.

Reply:
replaced with a for loop: https://github.com/fabricjs/fabric.js/pull/6013/commits/dbaf781eb48feeaf32a96845daa6657f18d457c1